### PR TITLE
feat(feature-activation): implement bit signal in block templates

### PR DESCRIPTION
**Pull request by glevco**
### Motivation

What was the motivation for the changes in this PR?

### Acceptance Criteria

- Include here all things that this PR should solve

### Checklist

- [ ] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

**Review by jansegre — APPROVED** [review]

File: hathor/simulator/miner/geometric_miner.py [hunk 1]
```diff
@@ -75,7 +75,10 @@ def _generate_mining_block(self) -> 'Block':
         except IndexError:
             signal_bits = 0
 
-        return self._manager.generate_mining_block(signal_bits=signal_bits)
+        block = self._manager.generate_mining_block()
+        block.signal_bits = signal_bits
```
Comment (msbrogli) [inline hathor/simulator/miner/geometric_miner.py:79]: Shouldn't we use a `BitSignalingService` here as well?

File: hathor/transaction/base_transaction.py [hunk 2]
```diff
@@ -1035,6 +1035,7 @@ def to_json(self, decode_script: bool = False, include_metadata: bool = False) -
         data['timestamp'] = self.timestamp
         data['version'] = int(self.version)
         data['weight'] = self.weight
+        data['signal_bits'] = self.signal_bits
```
Comment (msbrogli) [inline hathor/transaction/base_transaction.py:1038]: Should we serialize it as a string of zeros and ones?

**Review by msbrogli — APPROVED** [review]

File: hathor/simulator/miner/geometric_miner.py (hunk 1 above)
Comment (glevco) [reply hathor/simulator/miner/geometric_miner.py:79]: The `BitSignalingService` generates a "fixed" set of signal bits that is the same for all generated blocks (assuming the feature states do not change). Conversely, in [some tests](https://github.com/HathorNetwork/hathor-core/blob/master/tests/feature_activation/test_feature_service.py#L35-L68) I define a different signal bits combination for each block, to test specific scenarios, using the `GeometricMiner`.

In other words, the signal bits are mocked manually here in the `GeometricMiner`, since the `BitSignalingService` doesn't allow such behavior.

File: hathor/transaction/base_transaction.py (hunk 2 above)
Comment (glevco) [reply hathor/transaction/base_transaction.py:1038]: I don't see a reason why, what are your thoughts?

Comment (msbrogli) [reply hathor/transaction/base_transaction.py:1038]: I feel it's weird to get an integer number. But it's ok too.

File: hathor/simulator/miner/geometric_miner.py (hunk 1 above)
Comment (msbrogli) [reply hathor/simulator/miner/geometric_miner.py:79]: Got it. Thanks for the clarification.
